cookie_monster/static/cookie_monster/js/base.js

```diff
@@ -1,86 +1,115 @@
+     * Read cookies in the browser
+     * @return {boolean}
+     */
+    function hasAcceptedCookies() {
+        if (!document.cookie) {
+            return false;
+        }
 
-// Read cookies in the browser
-function checkCookies() {
-    var cookieList = (document.cookie) ? document.cookie.split(/[;\s=]+/).indexOf(window.__js_base_cookie_bridge.cookie_identifier) : [];
-    return cookieList !== -1;
-}
+        var cookies = document.cookie.split(/[;\s=]+/);
+        return cookies.indexOf(COOKIE_IDENTIFIER) !== -1;
+    }
 
-// Push the checked cookies into an array
-function getCheckedCookies() {
-    document.querySelectorAll('.cookieMakerContainerContent input[type="checkbox"]:checked').forEach(function (el) {
-        return cookieValues.push(' ' + el.value);
-    })
-}
+    /**
+     * Get selected cookies
+     * @return {string[]}
+     */
+    function getCheckedCookies() {
+        return Array.from(document.querySelectorAll('.' + BANNER_CLASSNAME + ' input[type="checkbox"]:checked')).map(function(el) {
+            return el.value;
+        })
+    }
 
-// Push all cookies into an array
-function selectAllCookies() {
-    document.querySelectorAll('.cookieMakerContainerContent input[type="checkbox"]').forEach(function (el) {
-        return cookieValues.push(' ' + el.value);
-    })
-}
+    /**
+     * Get all available cookies
+     * @return {string[]}
+     */
+    function getAllCookies() {
+        return Array.from(document.querySelectorAll('.' + BANNER_CLASSNAME + ' input[type="checkbox"]')).map(function(el) {
+            return el.value;
+        })
+    }
 
-// Logic to create the cookie
-function createCookie() {
-    var expires = "";
-    var config = window.__js_base_cookie_bridge;
-    if (config.cookie_age) {
-        var date = new Date();
-        date.setTime(date.getTime() + (config.cookie_age * 365 * 24 * 60 * 60 * 1000));
-        expires = "; expires=" + date.toUTCString();
+    /**
+     * Create cookie with selected values
+     * @param {string} identifier
+     * @paramn {string[]|null} values
+     */
+    function createCookie(identifier, values) {
+        var date = new Date((new Date).getTime() + (COOKIE_AGE * 1000));
+        var cookieVal = (Array.isArray(values) ? values : true);
+        var cookieValue = identifier + '=' + cookieVal;
+        cookieValue += '; expires=' + date.toUTCString();
+        cookieValue += '; path=/;';
+        document.cookie = cookieValue;
     }
-    document.cookie = config.cookie_identifier + "=" + ((cookieValues.length > 0) ? cookieValues : true) + expires + "; path=/";
-}
 
-// Fade banner functions
-function cookieBannerFadeIn(elem, display) {
-    var el = document.getElementById(elem);
-    el.style.opacity = 0;
-    el.style.display = display || "block";
+    /**
+     * Set the banner state
+     * @param {string} elem
+     * @param {boolean} show
+     */
+    function activateBanner(elem, show) {
+        var el = document.querySelectorAll(elem);
```

Comment:
here you forgot to create an Array from a NodeList. You need to fix it because it will not work on older browsers